File: docs/sdks/typescript/smart-cities/examples/pagination.ts
/**
 * Smart Cities SDK - Pagination Example
 *
 * This example shows how to paginate through large result sets
 */

import { createClient } from '@lydian/smart-cities-sdk';

const client = createClient({
  auth: { apiKey: process.env.LYDIAN_API_KEY! }
});

// Paginate through all cities
let hasMore = true;
let cursor: string | undefined;
let allCities = [];

while (hasMore) {
  const response = await client.listCities({ cursor, limit: 50 });
  allCities.push(...response.data);
  cursor = response.nextCursor;
  hasMore = response.hasMore;
}

console.log(`Total cities: ${allCities.length}`);

File: docs/sdks/python/lydian-smart-cities/examples/pagination.py
"""
Smart Cities SDK - Pagination Example

This example shows how to paginate through large result sets
"""

import os
from lydian_smart_cities import SmartCitiesClient

client = SmartCitiesClient(api_key=os.getenv("LYDIAN_API_KEY"))

# Paginate through all cities
all_cities = []
cursor = None

while True:
    response = client.list_cities(cursor=cursor, limit=50)
    all_cities.extend(response.data)

    if not response.has_more:
        break

    cursor = response.next_cursor

print(f"Total cities: {len(all_cities)}")
